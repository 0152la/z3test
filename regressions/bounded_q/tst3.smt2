
(declare-fun r (Int Int) Bool)
(declare-fun a () Int)

(assert (forall ((x Int))
                (=> (and (<= 0 x) (<= x 3))
                    (forall ((y Int))
                            (=> (and (<= 0 y) (<= y x))
                                (r x y))))))

(apply (then (! simplify :arith-lhs true :elim-and true) nnf normalize_bounded_quantifiers (! expand_bounded_quantifiers :domain 100)))

