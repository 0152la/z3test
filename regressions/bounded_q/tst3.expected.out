(goals
(goal
  (r 0 0)
  (r 1 0)
  (r 1 1)
  (r 2 0)
  (r 2 1)
  (r 2 2)
  (r 3 0)
  (r 3 1)
  (r 3 2)
  (r 3 3)
  :precision precise :depth 5)
)
